File: hier/project-euler/euler-001-hackerrank/euler001.cs
using static System.Console;

[assembly: System.Reflection.AssemblyVersion("1.0.0.0")]

internal static class Solution {
    private static long Sum(long to) => (to - 1L) * to / 2L;

    private static long Sum(long to, long by)
            => Sum(to / by + (to % by == 0L ? 0L : 1L)) * by;

    private static long Sum(long to, long by1, long by2)
            => Sum(to, by1) + Sum(to, by2) - Sum(to, by1 * by2);

    private static void Main()
    {
        const long a = 3L, b = 5L; // factors

        for (var t = int.Parse(ReadLine()); t > 0; --t)
            WriteLine(Sum(long.Parse(ReadLine()), a, b));
    }
}

File: hier/project-euler/euler-001-hackerrank/euler001.java
final class Solution {
    private static long sum(final long to)
    {
        return (to - 1L) * to / 2L;
    }
    
    private static long sum(final long to, final long by)
    {
        return sum(to / by + (to % by == 0L ? 0L : 1L)) * by;
    }
    
    private static long sum(final long to, final long by1, final long by2)
    {
        return sum(to, by1) + sum(to, by2) - sum(to, by1 * by2);
    }
    
    public static void main(final String[] args)
    {
        final long a = 3L, b = 5L; // factors
        
        try (final java.util.Scanner sc = new java.util.Scanner(System.in)) {
            for (int t = sc.nextInt(); t > 0; --t)
                System.out.println(sum(sc.nextLong(), a, b));
        }
    }
    
    private Solution() { } // the Solution class should not be instantiated
}
